python/ray/dashboard/metrics_exporter/schema.py

```diff
@@ -8,13 +8,27 @@ class ValidationError(Exception):
 class BaseModel:
     """Base class to define schema.
 
-    This will raise ValidationError if
-    - Number of given kwargs are bigger than needed.
-    - Number of given kwargs are smaller than needed.
-
-    This doesn't
-    - Validate types.
+    Model schema should be defined in class variable `__schema__`
+    within a child class. `__schema__` should be a dictionary that
+    contains `field`: `(required: bool, default: Any, type: type)`
+    See the example below for more details.
+
+    The class can have unexpected behavior if you don't follow the
+    schema pattern properly.
+
+    Example:
+        class A(BaseModel):
+            __schema__ = {
+                #             (REQUIRED,     DEFAULT,   TYPE)
```

Comment:
required and default are mutually exclusive. so these fields shouldn't be separate field. Here's a suggestion:
```
REQUIRED = object()

__schame__ = {
    "field": (REQUIRED, type),
    "other": (23, int),
}
```
Maybe this is better?

Reply:
Just to be clear, REQUIRED here is a singleton. https://python-patterns.guide/python/sentinel-object/

Reply:
Alternatively, named tuple can help readability
https://docs.python.org/3/library/collections.html#collections.namedtuple
```
"field": attr(required=False, default=33, type=int)
```

Reply:
I like that style. Will fix it!